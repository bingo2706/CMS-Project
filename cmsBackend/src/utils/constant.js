export const TYPE_ROLE = {
    ROLE_ADMIN: 'R1',
    ROLE_MEMBER: 'R2',
};
