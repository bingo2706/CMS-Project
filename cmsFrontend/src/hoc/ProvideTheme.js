import { CssBaseline, ThemeProvider } from '@mui/material';
import { ColorModeContext, useMode } from '../mockTheme';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { BrowserRouter } from 'react-router-dom';
import { store, persistor } from '../redux/store';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';
function ProvideTheme(props) {
    const [theme, colorMode] = useMode();
    return (
        <BrowserRouter>
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <LocalizationProvider dateAdapter={AdapterDayjs}>
                        <ColorModeContext.Provider value={colorMode}>
                            <ThemeProvider theme={theme}>
                                <CssBaseline />
                                {props.children}
                            </ThemeProvider>
                        </ColorModeContext.Provider>
                    </LocalizationProvider>
                </PersistGate>
            </Provider>
        </BrowserRouter>
    );
}
export default ProvideTheme;
