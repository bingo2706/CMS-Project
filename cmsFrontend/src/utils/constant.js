export const LANGUAGE = {
    CHANGE_LANGUAGE_START: 'CHANGE_LANGUAGE_START',
    CHANGE_LANGUAGE_SUCCESS: 'CHANGE_LANGUAGE_SUCCESS',
};
export const PROJECT = {
    GET_ALL_PROJECT_START: 'GET_ALL_PROJECT_START',
    GET_ALL_PROJECT_PENDING: 'GET_ALL_PROJECT_PENDING',
    GET_ALL_PROJECT_SUCCESS: 'GET_ALL_PROJECT_SUCCESS',
};
export const USER = {
    LOGIN_START: 'LOGIN_START',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',

};
export const FORMAT = {
    FORMAR_DATE: 'DD/MM/YYYY',
};
export const TYPE_ALLCODE = {
    ROLE: 'ROLE',
    GENDER: 'GENDER',
    STATUSPROJECT: 'STATUSPROJECT',
};
export const TYPE_LOGIN = {
    TRADITIONAL: 'traditional',
    OAUTH: 'oauth',
    TYPE_LOGIN: 'typeLogin',
    USER_DATA: 'userData',
    ROLE_ADMIN: 'R1',
};
export const FLAG_URL = {
    VI: 'https://upload.wikimedia.org/wikipedia/commons/thumb/2/21/Flag_of_Vietnam.svg/1280px-Flag_of_Vietnam.svg.png',
    EN: 'https://upload.wikimedia.org/wikipedia/en/thumb/a/ae/Flag_of_the_United_Kingdom.svg/800px-Flag_of_the_United_Kingdom.svg.png',
};
export const USER_REFERENCE = {
    USER_REFERENCE: 'userReference',
    TABLE_USER: 'tableUser',
    TABLE_PROJECT: 'tableProject',
};
