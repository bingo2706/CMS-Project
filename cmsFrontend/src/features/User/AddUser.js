import { Box, Button, TextField } from '@mui/material';
import Stack from '@mui/material/Stack';
import useMediaQuery from '@mui/material/useMediaQuery';
import Header from '../../components/common/HeaderList';
import Autocomplete from '@mui/material/Autocomplete';
import { useFetchAllcode } from '../../hooks/fetchAllcode';
import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';
import { useState } from 'react';
import CommonUtils from '../../utils/CommonUtils';
import PhotoCamera from '@mui/icons-material/PhotoCamera';
import Lightbox from 'react-image-lightbox';

// import 'react-image-lightbox/style.css';
import { createNewUser } from '../../services/userService';
import { toast } from 'react-toastify';
import { FORMAT, TYPE_ALLCODE } from '../../utils/constant';
function AddUser() {
    const isNonMobile = useMediaQuery('(min-width:600px)');
    var { data: dataRole } = useFetchAllcode(TYPE_ALLCODE.ROLE);
    var { data: dataGender } = useFetchAllcode(TYPE_ALLCODE.GENDER);
    const [inputValues, setInputValues] = useState({
        image: '',
        imageReview: '',
        isOpen: false,
        firstName: '',
        lastName: '',
        email: '',
        phonenumber: '',
        genderId: '',
        roleId: '',
        password: '',
        date: new Date(),
        genderOptions: '',
        roleOptions: '',
    });

    const handleOnChange = (event) => {
        const { name, value } = event.target;
        setInputValues({ ...inputValues, [name]: value });
    };
    let handleOnChangeImage = async (event) => {
        let data = event.target.files;
        let file = data[0];
        if (file.size > 31312281) {
            alert('Dung lượng file bé hơn 30mb');
        } else {
            console.log('file', file);
            let base64 = await CommonUtils.getBase64(file);
            let objectUrl = URL.createObjectURL(file);
            setInputValues({ ...inputValues, image: base64, imageReview: objectUrl });
        }
    };
    let openPreviewImage = () => {
        console.log('click');
        if (!inputValues.imageReview) return;

        setInputValues({ ...inputValues, isOpen: true });
    };
    const handleSaveUser = async () => {
        try {
            let res = await createNewUser({
                email: inputValues.email,
                password: inputValues.password,
                firstName: inputValues.firstName,
                lastName: inputValues.lastName,
                roleId: inputValues.roleId,
                genderId: inputValues.genderId,
                phonenumber: inputValues.phonenumber,
                avatar: inputValues.image,
                dob: inputValues.date,
            });
            if (res && res.errCode === 0) {
                toast.success('Lưu người dùng thành công');
                setInputValues({
                    ...inputValues,
                    image: '',
                    imageReview: '',
                    isOpen: false,
                    firstName: '',
                    lastName: '',
                    email: '',
                    phonenumber: '',
                    genderId: '',
                    roleId: '',
                    password: '',
                    date: new Date(),
                    genderOptions: '',
                    roleOptions: '',
                });
            } else {
                toast.error(res.errMessage);
            }
        } catch (error) {
            toast.error(error.response.data.errMessage);
        }
    };
    return (
        <Box m="20px">
            <Header title="CREATE USER" subtitle="Create a New User Profile" />

            <Box
                display="grid"
                gap="30px"
                gridTemplateColumns="repeat(4, minmax(0, 1fr))"
                sx={{
                    '& > div': { gridColumn: isNonMobile ? undefined : 'span 4' },
                }}
            >
                <TextField
                    fullWidth
                    variant="filled"
                    type="text"
                    label="First Name"
                    onChange={(e) => handleOnChange(e)}
                    value={inputValues.firstName}
                    name="firstName"
                    color="secondary"
                    sx={{ gridColumn: 'span 2' }}
                />
                <TextField
                    fullWidth
                    variant="filled"
                    type="text"
                    label="Last Name"
                    onChange={(e) => handleOnChange(e)}
                    value={inputValues.lastName}
                    name="lastName"
                    color="secondary"
                    sx={{ gridColumn: 'span 2' }}
                />
                <TextField
                    fullWidth
                    variant="filled"
                    type="text"
                    label="Email"
                    onChange={(e) => handleOnChange(e)}
                    value={inputValues.email}
                    name="email"
                    color="secondary"
                    sx={{ gridColumn: 'span 2' }}
                />
                <TextField
                    fullWidth
                    variant="filled"
                    type="password"
                    label="Password"
                    onChange={(e) => handleOnChange(e)}
                    value={inputValues.password}
                    name="password"
                    color="secondary"
                    sx={{ gridColumn: 'span 2' }}
                />

                <Autocomplete
                    disablePortal
                    onChange={(event, newValue) => {
                        setInputValues({ ...inputValues, genderId: newValue.value, genderOptions: newValue });
                    }}
                    value={inputValues.genderOptions}
                    options={dataGender}
                    sx={{ gridColumn: 'span 2' }}
                    renderInput={(params) => <TextField color="secondary" fullWidth variant="filled" {...params} label="Gender" />}
                />
                <Autocomplete
                    disablePortal
                    onChange={(event, newValue) => {
                        setInputValues({ ...inputValues, roleId: newValue.value, roleOptions: newValue });
                    }}
                    value={inputValues.roleOptions}
                    options={dataRole}
                    sx={{ gridColumn: 'span 2' }}
                    renderInput={(params) => <TextField color="secondary" fullWidth variant="filled" {...params} label="Role" />}
                />

                <Stack sx={{ gridColumn: 'span 2' }} spacing={3}>
                    <DesktopDatePicker
                        label="Date of birth"
                        inputFormat={FORMAT.FORMAR_DATE}
                        value={inputValues.date}
                        name="date"
                        onChange={(value) => setInputValues({ ...inputValues, date: value['$d'] })}
                        renderInput={(params) => <TextField color="secondary" fullWidth variant="filled" {...params} />}
                    />
                </Stack>

                <TextField
                    fullWidth
                    variant="filled"
                    type="text"
                    label="Phonenumber"
                    onChange={(e) => handleOnChange(e)}
                    value={inputValues.phonenumber}
                    name="phonenumber"
                    color="secondary"
                    sx={{ gridColumn: 'span 2' }}
                />
            </Box>
            <Box display="flex" alignItems="flex-start" flexDirection="column" gap="10px" mt="10px">
                <span id="title">Avatar</span>
                <img
                    data-testid="imgReview"
                    onClick={() => openPreviewImage()}
                    src={inputValues.image}
                    style={{ width: '100px', height: '100px', objectFit: 'cover', cursor: 'pointer', borderRadius: 5 }}
                    alt="avatar"
                />
                <Box display={'flex'} gap="10px" alignItems={'center'}>
                    <Button variant="contained" component="label" color="secondary">
                        <PhotoCamera sx={{ marginRight: 1 }} />
                        Upload
                        <input hidden accept="image/*" onChange={(event) => handleOnChangeImage(event)} type="file" />
                    </Button>
                    <Button onClick={() => handleSaveUser()} type="submit" color="neutral" variant="contained">
                        Create New User
                    </Button>
                </Box>
            </Box>

            {inputValues.isOpen === true && (
                <Lightbox
                    data-testid="Lightbox"
                    mainSrc={inputValues.imageReview}
                    onCloseRequest={() => setInputValues({ ...inputValues, isOpen: false })}
                />
            )}
        </Box>
    );
}

export default AddUser;
