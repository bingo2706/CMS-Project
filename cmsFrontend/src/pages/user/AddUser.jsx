import AddUser from '../../features/User/AddUser';
import { withLayoutAdmin } from '../../hoc/withLayoutAdmin';

function AddUSer() {
    return <AddUser />;
}
export default withLayoutAdmin(AddUSer);
