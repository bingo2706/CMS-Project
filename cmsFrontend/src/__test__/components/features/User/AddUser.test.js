import React from 'react';
import AddUser from '../../../../features/User/AddUser';
import { fireEvent, render } from '@testing-library/react';
import ProvideTheme from '../../../../hoc/ProvideTheme';
import { wait } from '@testing-library/user-event/dist/utils';
import TestRenderer from 'react-test-renderer';
describe('Add User component', () => {
    it('should be match snapshot', () => {
        const { container } = render(
            <ProvideTheme>
                <AddUser />,
            </ProvideTheme>,
        );
        expect(container).toMatchSnapshot();
    });
    it('openReviewImage', async () => {
        const { container, getByTestId } = render(
            <ProvideTheme>
                <AddUser />,
            </ProvideTheme>,
        );
        const imgReview = getByTestId('imgReview');

        fireEvent.click(imgReview);
        await wait(() => {
            const Lightbox = getByTestId('Lightbox');
            expect(Lightbox).toBeDefined();
        });
    });
    it('onchangeInput', () => {
        const { container, getAllByTestId, getByLabelText } = render(
            <ProvideTheme>
                <AddUser />,
            </ProvideTheme>,
        );
        const inputFirstName = getByLabelText('First Name');
        fireEvent.change(inputFirstName, { target: { value: '123', name: 'First Name' } });
    });
});
