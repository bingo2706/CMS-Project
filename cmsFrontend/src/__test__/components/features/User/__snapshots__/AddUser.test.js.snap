// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add User component should be match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-zwiamn"
  >
    <div
      class="MuiBox-root css-1xn3edc"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-uv7eon-MuiTypography-root"
        data-testid="title"
        id="title"
      >
        CREATE USER
      </h2>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ph60q2-MuiTypography-root"
        data-testid="subtitle"
        id="subtitle"
      >
        Create a New User Profile
      </h5>
    </div>
    <div
      class="MuiBox-root css-1q4o5na"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vd3lbb-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          First Name
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-18y3qso-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
            id=":r0:"
            name="firstName"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vd3lbb-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Last Name
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-18y3qso-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
            id=":r1:"
            name="lastName"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vd3lbb-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r2:"
          id=":r2:-label"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-18y3qso-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
            id=":r2:"
            name="email"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vd3lbb-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r3:"
          id=":r3:-label"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-18y3qso-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
            id=":r3:"
            name="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-zwpqmf-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r4:"
            id=":r4:-label"
          >
            Gender
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-tz8f39-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1a4pl44-MuiInputBase-input-MuiFilledInput-input"
              id=":r4:"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1cleyyo-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-196n7va-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-10sixfi-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-havevq-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-zwpqmf-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r6:"
            id=":r6:-label"
          >
            Role
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-tz8f39-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1a4pl44-MuiInputBase-input-MuiFilledInput-input"
              id=":r6:"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1cleyyo-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-196n7va-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-10sixfi-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-havevq-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1gwwi9r-MuiStack-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorSecondary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-hkp1b2-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r8:"
            id=":r8:-label"
          >
            Date of birth
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-tz8f39-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd css-1a4pl44-MuiInputBase-input-MuiFilledInput-input"
              id=":r8:"
              placeholder="dd/mm/yyyy"
              type="tel"
              value="09/01/2023"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-1cuuaz0-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is Jan 9, 2023"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1y8qrae-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-havevq-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vd3lbb-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled css-1827exq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r9:"
          id=":r9:-label"
        >
          Phonenumber
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-18y3qso-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
            id=":r9:"
            name="phonenumber"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-5tomlp"
    >
      <span
        id="title"
      >
        Avatar
      </span>
      <img
        alt="avatar"
        data-testid="imgReview"
        src=""
        style="width: 100px; height: 100px; object-fit: cover; cursor: pointer; border-radius: 5px;"
      />
      <div
        class="MuiBox-root css-15zum9u"
      >
        <label
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wnx1cx-MuiButtonBase-root-MuiButton-root"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i2rnez-MuiSvgIcon-root"
            data-testid="PhotoCameraIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="3.2"
            />
            <path
              d="M9 2 7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
            />
          </svg>
          Upload
          <input
            accept="image/*"
            hidden=""
            type="file"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </label>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedNeutral MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedNeutral MuiButton-sizeMedium MuiButton-containedSizeMedium css-1d6wr33-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Create New User
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  ,
</div>
`;
